src/software/full_system_main.cpp

```diff
@@ -94,11 +95,29 @@ int main(int argc, char **argv)
             backend->Subject<SensorProto>::registerObserver(visualizer);
         }
 
-        if (!args->replay_output_dir()->value().empty())
+        // TODO: manage this mess better and don't log SensorMsg when we are replaying
+        // TODO: make a shim between SensorFusion and ProtoLogger to log filtered world
```

Comment:
sorry what would this shim do?

Reply:
it would be an `Observer<World>` and `Subject<TbotsProto::Vision>` and do the conversion

Reply:
That's pretty close to what a `Backend` does

Reply:
Should we have backend connect `Subject`s to `ProtoLogger`s? I don't know if we'd want `Backend` to become a place to arbitrarily throw communication code

Reply:
It would be good idea to take the time to plan out an architecture before diving right in

Reply:
Rescoped the PR to only concern logging `PrimitiveSet`, logging filtered world data is gonna take some more design. Opened #1789 for this